firmware/application/apps/ui_looking_glass_app.cpp

```diff
@@ -34,11 +35,39 @@ void GlassView::focus() {
 }
 
 GlassView::~GlassView() {
+    audio::output::stop();
     receiver_model.set_sampling_rate(3072000);  // Just a hack to avoid hanging other apps
     receiver_model.disable();
     baseband::shutdown();
 }
 
+// Function to map the value from one range to another
+int32_t GlassView::map(int32_t value, int32_t fromLow, int32_t fromHigh, int32_t toLow, int32_t toHigh) {
+    return toLow + (value - fromLow) * (toHigh - toLow) / (fromHigh - fromLow);
+}
+
+void GlassView::update_display_beep() {
+    if (beep_enabled) {
+        button_beep_squelch.set_style(&Styles::green);
+        // <bip:-XXXdb>
+        button_beep_squelch.set_text("[bip>" + to_string_dec_int(beep_squelch, 4) + "db]");
+        receiver_model.set_headphone_volume(receiver_model.headphone_volume());  // WM8731 hack.
+    } else {
+        button_beep_squelch.set_style(&Styles::white);
+        button_beep_squelch.set_text("[ beep OFF [");
+    }
+}
+
+void GlassView::manage_beep_audio() {
+    if (beep_enabled) {
+        audio::set_rate(audio::Rate::Hz_48000);
```

Comment:
Same comment as earlier.  Audio sampling frequency setting here doesn't seem to match the sampling frequency in the call to request_audio_beep.

Reply:
Fixed